sdk/dapp-kit/src/components/WalletProvider.tsx

```diff
@@ -93,6 +95,28 @@ export function WalletProvider({
 
 	useUnsafeBurnerWallet(enableUnsafeBurner);
 
+	useEffect(() => {
+		if (!autoConnect || walletState.currentWallet) {
+			return;
+		}
+
+		(async function autoConnectWallet() {
+			const connectionInfo = await getMostRecentWalletConnectionInfo(storageAdapter, storageKey);
+			const { walletName, accountAddress } = connectionInfo || {};
+			const wallet = walletName
+				? walletState.wallets.find((wallet) => wallet.name === walletName)
+				: null;
+
+			if (wallet) {
+				connectWallet(dispatch, storageAdapter, storageKey, {
+					wallet,
+					accountAddress,
+					silent: true,
+				});
+			}
+		})();
```

Comment:
While maybe technically more proper, I came to the conclusion that this effect doesn't really _need_ a clean-up function (i.e. disconnect) since the `walletState.currentWallet` will always prevent double connection attempts. Curious if y'all have any thoughts on this

Reply:
I think in practice you're right (autoconnection will always happen on mount, and will be _near-synchronous_, whereas the explicit connection attempts will always happen far after that (or at least far from the perspective of CPU cycles)). I think technically there's one race condition where you don't have the currentWallet yet set because the `connect` feature is async, so you could technically kick two of these off. But honestly I am not really worried about it, more just academically thinking about it (you could also prevent this using the connection status although I also think it's fine to remove that tbh and just let folks race this because you have to go out your way to make it happen)